tests/v2/integration/catalogv2/system_charts_version_test.go

```diff
@@ -0,0 +1,272 @@
+	client               *rancher.Client
+	session              *session.Session
+	restClientGetter     genericclioptions.RESTClientGetter
+	catalogClient        *catalog.Client
+	latestWebhookVersion string
+}
+
+func (w *SystemChartsVersionSuite) TearDownSuite() {
+	_ = w.updateSetting("rancher-webhook-version", w.latestWebhookVersion)
+	_ = w.updateSetting("rancher-webhook-min-version", "")
+	_ = w.updateSetting("system-feature-chart-refresh-seconds", "3600")
+	w.session.Cleanup()
+}
+
+func (w *SystemChartsVersionSuite) SetupSuite() {
+	var err error
+	testSession := session.NewSession()
+	w.session = testSession
+
+	w.client, err = rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+
+	w.catalogClient, err = w.client.GetClusterCatalogClient("local")
+	require.NoError(w.T(), err)
+
+	host := hostName(w.client.RestConfig.Host)
+	w.restClientGetter = &genericclioptions.ConfigFlags{
+		CAFile:      &w.client.RestConfig.TLSClientConfig.CAFile,
+		APIServer:   &host,
+		Insecure:    &w.client.RestConfig.TLSClientConfig.Insecure,
+		BearerToken: &w.client.RestConfig.BearerToken,
+	}
+
+	w.latestWebhookVersion, err = w.catalogClient.GetLatestChartVersion("rancher-webhook")
+	require.NoError(w.T(), err)
+
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-version", w.latestWebhookVersion))
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-min-version", ""))
+	require.NoError(w.T(), w.updateSetting("system-feature-chart-refresh-seconds", "10"))
+}
+
+func (w *SystemChartsVersionSuite) resetSettings() {
+	w.T().Helper()
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-version", w.latestWebhookVersion))
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-min-version", ""))
+	require.NoError(w.T(), w.updateSetting("system-feature-chart-refresh-seconds", "10"))
+}
+
+func TestSystemChartsVersionSuite(t *testing.T) {
+	suite.Run(t, new(SystemChartsVersionSuite))
+}
+
+func (w *SystemChartsVersionSuite) TestInstallWebhook() {
+	defer w.resetSettings()
+
+	const exactVersion = "2.0.3+up0.3.3"
+	w.Require().NoError(w.uninstallApp("cattle-system", "rancher-webhook"))
+	w.Require().NoError(w.updateSetting("rancher-webhook-version", exactVersion))
+
+	watcher, err := w.catalogClient.Apps("cattle-system").Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + "rancher-webhook",
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	w.Require().NoError(err)
+
+	err = wait.WatchWait(watcher, func(event watch.Event) (ready bool, err error) {
+		if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error installing the rancher-webhook chart")
+		} else if event.Type == watch.Added {
+			return true, nil
+		}
+		return false, nil
+	})
+	w.Require().NoError(err)
+
+	// Allow the new release to fully deploy. Otherwise, the client won't find it among current releases.
+	var newRelease *release.Release
+	err = kwait.Poll(10*time.Second, time.Minute, func() (done bool, err error) {
```

Comment:
It might be good to define the timeout of this in a constant at the top of the file, and include the timeout in the error message if things don't install in the right time. I'm thinking that the "if you change the version a new webhook installs within 'x'seconds" could be part of our functional contract.